python/tk_framework_desktopserver/server.py

```diff
@@ -98,8 +108,15 @@ def start(self, start_reactor=False):
         if start_reactor:
             self._start_reactor()
 
-    def stop(self):
+    def is_running(self):
+        """
+        :returns: True if the server is up and running, False otherwise.
+        """
+        return self._reactor_thread.isAlive()
+
+    def tear_down(self):
         """
-        Stops the serverreactor.
+        Tears down Twisted.
         """
         reactor.callFromThread(reactor.stop)
+        self._reactor_thread.join()
```

Comment:
When is start_reactor false in the args to start?

Given the current code if it is false, then I think self._reactor_thread won't be defined when tear_down is called and this will error.